Common/Product/SharedProject/UIThread.cs

```diff
@@ -47,6 +49,23 @@ class UIThread : UIThreadBase {
             }
         }
 
+        public void Dispose() {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        ~UIThread() {
```

Comment:
There are no unmanaged resources, you don't need a destructor.

Reply:
Maybe, but this is to keep static analysis from complaining, so I'm covering all bases.